pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -87,3 +91,228 @@ func (isvc *InferenceService) DefaultInferenceService(config *InferenceServicesC
 		}
 	}
 }
+
+func (isvc *InferenceService) setPredictorModelDefaults() {
+	switch {
+	case isvc.Spec.Predictor.SKLearn != nil:
+		isvc.assignSKLearnRuntime()
+
+	case isvc.Spec.Predictor.Tensorflow != nil:
+		isvc.assignTensorflowRuntime()
+
+	case isvc.Spec.Predictor.XGBoost != nil:
+		isvc.assignXGBoostRuntime()
+
+	case isvc.Spec.Predictor.PyTorch != nil:
+		isvc.assignPyTorchRuntime()
+
+	case isvc.Spec.Predictor.Triton != nil:
+		isvc.assignTritonRuntime()
+
+	case isvc.Spec.Predictor.ONNX != nil:
+		isvc.assignONNXRuntime()
+
+	case isvc.Spec.Predictor.PMML != nil:
+		isvc.assignPMMLRuntime()
+
+	case isvc.Spec.Predictor.LightGBM != nil:
+		isvc.assignLightGBMRuntime()
+
+	case isvc.Spec.Predictor.Paddle != nil:
+		isvc.assignPaddleRuntime()
+	}
+}
+
+func (isvc *InferenceService) assignSKLearnRuntime() {
+	// skips if the storage uri is not specified
+	if isvc.Spec.Predictor.SKLearn.StorageURI == nil {
+		isvc.Spec.Predictor.Model = nil
+		return
+	}
+	// assign built-in runtime based on protocol version
+	var runtime = constants.SKLearnServer
+	if isvc.Spec.Predictor.SKLearn.ProtocolVersion != nil &&
+		constants.ProtocolV2 == *isvc.Spec.Predictor.SKLearn.ProtocolVersion {
+		runtime = constants.MLServer
+		if isvc.ObjectMeta.Labels == nil {
+			isvc.ObjectMeta.Labels = map[string]string{constants.ModelClassLabel: constants.MLServerModelClassSKLearn}
+		} else {
+			isvc.ObjectMeta.Labels[constants.ModelClassLabel] = constants.MLServerModelClassSKLearn
+		}
+	}
+	isvc.Spec.Predictor.Model = &ModelSpec{
+		Framework:              v1alpha1.Framework{Name: constants.SupportedModelSKLearn},
+		PredictorExtensionSpec: isvc.Spec.Predictor.SKLearn.PredictorExtensionSpec,
+		Runtime:                &runtime,
+	}
+	// remove sklearn spec
+	isvc.Spec.Predictor.SKLearn = nil
+}
+
+func (isvc *InferenceService) assignTensorflowRuntime() {
+	// skips if the storage uri is not specified
+	if isvc.Spec.Predictor.Tensorflow.StorageURI == nil {
+		isvc.Spec.Predictor.Model = nil
+		return
+	}
+	// assign built-in runtime based on gpu config
+	var runtime = constants.TFServing
+	if utils.IsGPUEnabled(isvc.Spec.Predictor.Tensorflow.Resources) {
+		runtime = constants.TFServingGPU
```

Comment:
Since the GPU runtimes were removed, looks like we don't need these checks anymore.